config/src/main/java/org/springframework/security/config/annotation/web/configurers/SecurityContextConfigurer.java

```diff
@@ -83,6 +84,19 @@ public SecurityContextConfigurer<H> securityContextRepository(SecurityContextRep
 		return this;
 	}
 
+	/**
+	 * Specifies the shared {@link SecurityContextHolderStrategy} that is to be used
+	 * @param securityContextHolderStrategy the {@link SecurityContextHolderStrategy} to
+	 * use
+	 * @return the {@link SecurityContextConfigurer} for further customizations
+	 * @since 5.8
+	 */
+	public SecurityContextConfigurer<H> securityContextHolderStrategy(
+			SecurityContextHolderStrategy securityContextHolderStrategy) {
+		getBuilder().setSharedObject(SecurityContextHolderStrategy.class, securityContextHolderStrategy);
```

Comment:
I have a concern that the `HttpSecurity` and `AuthenticationPrincipalArgumentResolver` may have different `SecurityContextHoderStrategy` associated with them.

Reply:
We'll leave the DSL support out for now. It would be rather unexpected to use a different `SecurityContextHolderStrategy` for each filter chain. That said, if this turns out to be needed, the argument resolver could possibly be updated to look at the request attribute to get the security context.